src/main/java/com/novi/backend_spring_boot_tech_it_easy_controller/services/CIModuleService.java

```diff
@@ -0,0 +1,81 @@
+package com.novi.backend_spring_boot_tech_it_easy_controller.services;
+
+import com.novi.backend_spring_boot_tech_it_easy_controller.dtos.CIModuleDto;
+import com.novi.backend_spring_boot_tech_it_easy_controller.dtos.CIModuleInputDto;
+import com.novi.backend_spring_boot_tech_it_easy_controller.models.CIModule;
+import com.novi.backend_spring_boot_tech_it_easy_controller.repositories.CIModuleRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CIModuleService {
+
+    private final CIModuleRepository cIModuleRepository;
+
+    @Autowired
+    public CIModuleService(CIModuleRepository cIModuleRepository) {
+        this.cIModuleRepository = cIModuleRepository;
+    }
+
+    public List<CIModuleDto> getAllCIModules() {
+        List<CIModule> ciModulesList = cIModuleRepository.findAll();
+        List<CIModuleDto> cIModuleDtoList = new ArrayList<>();
+
+        for (CIModule cIModule : ciModulesList) {
+            CIModuleDto cIModuleDto = convertToDto(cIModule);
+            cIModuleDtoList.add(cIModuleDto);
+        }
+
+        return cIModuleDtoList;
+    }
+
+    public CIModuleDto getCIModuleById(Long id) {
+        CIModule cIModule = cIModuleRepository.findById(id).orElseThrow(() -> new RuntimeException("CIModule not found"));
+        return convertToDto(cIModule);
+    }
+
+    public CIModuleDto createCIModule(CIModuleInputDto cIModuleInputDto) {
+        CIModule cIModule = new CIModule();
+        cIModule.setName(cIModuleInputDto.getName());
```

Comment:
Je kunt mapping (het over zetten van data van de ene klasse naar de andere klasse) beter in en eigen mapper functie of mapper klasse doen. Dat voldoet beter aan de clean code en SOLID principes